### PR TITLE
update obs

### PR DESCRIPTION
**Pull request by sashrika**
https://issues.openmrs.org/browse/FM-108

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir/api/util/FHIRObsUtil.java [hunk 1]
```diff
@@ -330,4 +351,34 @@ public static Obs generateOpenMRSObs(Observation observation, List<String> error
...
+		Concept concept=requestObs.getConcept();
+		if (concept != null) {
+			if (requestObs.getConcept().isNumeric()) {
+				retrievedObs.setValueNumeric(requestObs.getValueNumeric());
+			} else if (FHIRConstants.ST_HL7_ABBREVATION.equalsIgnoreCase(concept.getDatatype().getHl7Abbreviation())) {
+				try {
+					retrievedObs.setValueAsString(requestObs.getValueAsString(Context.getLocale()));
+				}
+				catch (ParseException e) {
+					e.printStackTrace();
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/module/fhir/api/util/FHIRObsUtil.java]: no print stack traces. Use appropriate log.error or something

File: omod/src/main/java/org/openmrs/module/fhir/providers/RestfulObservationResourceProvider.java [hunk 2]
```diff
@@ -168,5 +169,26 @@ public MethodOutcome createFHIRObservation(@ResourceParam Observation observatio
...
+	@Update
+	public MethodOutcome updateFHIRObservation(@ResourceParam Observation observation, @IdParam IdDt theId) {
+		MethodOutcome retVal = new MethodOutcome();
+		OperationOutcome outcome = new OperationOutcome();
+		try {
+			observation = provider.updateFHIRObservation(observation, theId.getIdPart());
+		}
+		catch (Exception e) {
+			outcome.addIssue()
+			        //UPDATE THISSSSSSSSSSSSSSSSSSSSSSSSSSSS
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulObservationResourceProvider.java]: remove thise comments

Comment (sashrika) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulObservationResourceProvider.java]: @harsha89  , will address these.

File: api/src/main/java/org/openmrs/module/fhir/api/util/FHIRObsUtil.java [hunk 3]
```diff
@@ -223,9 +224,28 @@ public static Observation generateObs(Obs obs) {
...
+			Observation.Related related = new Observation.Related();
+			related.setType(ObservationRelationshipTypeEnum.REPLACES);
+			resourceReferenceDt.setDisplay("Old Obs which replaced by the new Obs");
+			IdDt providerRef = new IdDt();
+			String obsUri = FHIRConstants.OBSERVATION + "/" + obs.getPreviousVersion().getUuid();
+			providerRef.setValue(obsUri);
+			resourceReferenceDt.setReference(providerRef);
+			related.setTarget(resourceReferenceDt);
+			relatedObs.add(related);
+			//throw new NullPointerException();
```
Comment (sashrika) [inline api/src/main/java/org/openmrs/module/fhir/api/util/FHIRObsUtil.java]: remove comment

File: api/src/main/java/org/openmrs/module/fhir/api/impl/ObsServiceImpl.java [hunk 4]
```diff
@@ -243,4 +244,34 @@ public Observation createFHIRObservation(Observation observation) {
...
+		} else { // no observation is associated with the given uuid. so create a new observation with the given uuid
+			if (observation.getId() == null) { // since we need to PUT the observation to a specific URI, we need to set the uuid
+				// here, if it is not
+				// already set.
+				IdDt uuid = new IdDt();
+				uuid.setValue(theId);
+				observation.setId(uuid);
+			}
+			return createFHIRObservation(observation);
+		}
```
Comment (surangak) [inline api/src/main/java/org/openmrs/module/fhir/api/impl/ObsServiceImpl.java]: So if someone tries to update an existing obs, we do it, and return the new observation resource with the new UUID, as created by OpenMRS.
But if someone tries to update an obs that doesn't exist, we can't really assign the new UUID that comes in the FHIR request to the OpenMRS Obs object, right? we have to use whatever UUID created by the OpenMRS data model?

Comment (harsha89) [inline api/src/main/java/org/openmrs/module/fhir/api/impl/ObsServiceImpl.java]: Yeah if Ob is not existed with given uuid. We should refuse with resource not found.

Comment (harsha89) [inline api/src/main/java/org/openmrs/module/fhir/api/impl/ObsServiceImpl.java]: This hsould be fixed.

Comment (harsha89) [inline api/src/main/java/org/openmrs/module/fhir/api/impl/ObsServiceImpl.java]: After some discussion, we will keep this way. If no obs found. obs will create with given uuid.

**Pull request merged**